lib/common_test/src/ct_master.erl

```diff
@@ -140,10 +247,17 @@ exclude_nodes([],RunSkipPerNode) ->
 
 
 -doc """
-run_on_node(TestSpecs, AllowUserTerms, Node) -> ok
+run_on_node(TestSpecs, AllowUserTerms, Node) -> [{Specs, ok} | {error, Reason}]
 
 Tests are spawned on `Node` according to `TestSpecs`.
 """.
+-spec run_on_node(TestSpecs, AllowUserTerms, Node) -> [{Specs, 'ok'} | {'error', Reason}]
+              when TestSpecs :: TestSpec | [TestSpec] | [[TestSpec]],
+                   TestSpec :: test_spec(),
+                   AllowUserTerms :: boolean(),
+                   Node :: node(),
+                   Specs :: [file:filename_all()],
```

Comment:
Shouldn't these be Specs :: [module()] ??
And not filename  since filename is most often a "file_SUITE.erl"

Reply:
I don't think so since the specs in this list come from first tuple element from here:
https://github.com/Mikaka27/otp/blob/michal/common_test/specs/lib/common_test/src/ct_testspec.erl#L277C1-L300C9

And then if you follow create_testspecs function you can see that it calls create_specs which returns:
https://github.com/Mikaka27/otp/blob/michal/common_test/specs/lib/common_test/src/ct_testspec.erl#L391
And here you can see that it is some kind of path:
https://github.com/Mikaka27/otp/blob/michal/common_test/specs/lib/common_test/src/ct_testspec.erl#L365

Than this tuple goes here, and only first argument is returned:
https://github.com/Mikaka27/otp/blob/michal/common_test/specs/lib/common_test/src/ct_testspec.erl#L312

@u3s maybe you know more here?

Reply:
As far as I know these specs refer to files containing test run instructions. For instance ssl application has three spec files
in its test directory for instance ssl_bench.spec  for running benchmark tests looks like:


```
{alias, dir, "../ssl_test"}.
{groups, dir, ssl_dist_bench_SUITE, [benchmark]}.
{groups, dir, ssl_bench_SUITE,      [benchmark]}.

```

Reply:
@u3s  you had just a list with paths? Or tuples?

Reply:
list of strings being paths.

Reply:
Do we do anything with this now?

Reply:
Concluded on the scrum meeting, we leave it as is.